Support for TLS certificates and hostname verification

sberyozkin opened the pull request:
Fixes #571
Fixes #581

Currently only certificates in PEM format are supported and read from the file, but in the future we can support loading them from the truststores 

CC @kdubb

MikeEdgar commented on implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/AbstractKeyLocationResolver.java:
```diff
@@ -85,17 +91,31 @@ protected static String getKid(JsonWebStructure jws) {
+            }
+            if (trustedHosts != null) {
+                httpGet.setHostnameVerifier(new TrustedHostsHostnameVerifier(trustedHosts));
+            } else {
+                httpGet.setHostnameVerifier(new TrustAllHostnameVerifier());
+            }
```
Should the default be more defensive than trust all? I'm not sure :-)

sberyozkin replied:
Hi @MikeEdgar Yeah, good point, I think I'll need to introduce one more boolean property, `smallrye.jwt.tls.trust-all`, `false` is default

sberyozkin replied:
Mike, @MikeEdgar It should be better now, have a look please. Perhaps it can make sense for you to verify that the workaround you had to use is no longer necessary with this PR ?

kdubb replied:
I would expect this to verify hostnames by default but not require a restricted set of hostnames, unless trust-all is enabled.

MikeEdgar replied:
> Perhaps it can make sense for you to verify that the workaround you had to use is no longer necessary with this PR ?

LGTM - I'll try it as a replacement for the workaround probably early next week.

sberyozkin replied:
Thanks @MikeEdgar

MikeEdgar reviewed (approved):
LGTM, just one question.

sberyozkin commented:
@MikeEdgar I've simplified it quite a bit - as there was no need to update the `initializeHttpsJwks` signature with the need to update the code everywhere, and also added some basic mock tests, so it gets green then I'll merge, thanks

kdubb commented:
@sberyozkin Going by the documentation (and examining the code) with this PR there is no way to trust DNS host names, correct? It seems I can only disable host name checking (e.g. `trust-all`) or provide a set of allowed host names. Why not allow standard TLS semantics of checking DNS hostnames?

kdubb commented:
In our K8s environment we have TLS certs issued with valid CN’s and SAN’s matching DNS. This PR needlessly adds extra configuration overhead (explicitly setting hostnames) instead of relying on the standard TLS behavior.
